### PR TITLE
Use scope property to address nested resources

### PR DESCRIPTION
**Pull request by mdellweg**
This depends on https://github.com/pulp/pulpcore/pull/1066 to have proper href-keys for create of nested entities.
Until then only repository versions (that do not have "create") can benefit.

### CODE REVIEW

**Comment by mdellweg** [comment]
Let's figure out how to make this version dependent for exports in another PR.

**Review by daviddavis — APPROVED** [review]

File: pulpcore/cli/common/context.py [hunk 1]
```diff
@@ -169,13 +169,19 @@ class PulpEntityContext:
         "rpm": "rpm_rpm_repository_href",
     }
 
+    # Subclasses for nested entities can define the parameters for there parent scope here
+    @property
+    def scope(self) -> Dict[str, Any]:
```
Comment (daviddavis) [inline pulpcore/cli/common/context.py:174]: I just noticed this: would a `Dict[str, str]` make sense or is there a possibility that the value could be something other than a string?

Comment (mdellweg) [reply pulpcore/cli/common/context.py:174]: That's a good question. I can try making it `Dict[str, str]` and whenever mypy complains we can investigate why.

Comment (mdellweg) [reply pulpcore/cli/common/context.py:174]: Actually i just realized that a path parameter could also be an integer (like for repository versions). We don't nest anything beneath such an url yet, but it is a possibility. WDYT?

Comment (daviddavis) [reply pulpcore/cli/common/context.py:174]: Good point. `Dict[str, Any]` works then.

**Pull request merged**